app/models/team.rb

```diff
@@ -28,6 +28,11 @@ class Team < ApplicationRecord
   has_many :properties, class_name: TeamProperty, :dependent => :delete_all
```

Comment:
Can Properties also be eager loaded as that would ensure any calls to `team_lead` is not resource intensive

Reply:
I've eager loaded team_leader which is all that is required.